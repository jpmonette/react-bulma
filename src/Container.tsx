import * as React from "react";
import classNames from "classnames";

interface Props {
  className?: string;
  children?: React.ReactNode;
  isFluid?: boolean;
  isWidescreen?: boolean;
  isFullHD?: boolean;
}

class Container extends React.Component<Props> {
  render() {
    const { children, className, isFluid, isWidescreen, isFullHD, ...props } = this.props;

    return (
      <div
        {...props}
        className={classNames(
          "container",
          className,
          isFluid && "is-fluid",
          isWidescreen && "is-widescreen",
          isFullHD && "is-fullhd",
        )}
      >
        {children}
      </div>
    );
  }
}

export { Container };
