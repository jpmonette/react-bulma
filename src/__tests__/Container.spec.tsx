import * as React from "react";
import { Container } from "../Container";
import { create } from "react-test-renderer";

describe("Container", () => {
  it("should generate a container", () => {
    const tree = create(
      <Container isFluid isWidescreen isFullHD>
        Sample container
      </Container>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
