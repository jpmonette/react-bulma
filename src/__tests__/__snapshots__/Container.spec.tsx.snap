// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Container should generate a container 1`] = `
<div
  className="container is-fluid is-widescreen is-fullhd"
>
  Sample container
</div>
`;
